contrib/babelfishpg_tsql/src/iterative_exec.c

```diff
@@ -1306,13 +1306,16 @@ dispatch_stmt_handle_error(PLtsql_execstate *estate,
 			{
 				if (is_part_of_pltsql_trycatch_block(estate))
 				{
+					bool 	spi_current_internal_xact_mode;
+
 					HOLD_INTERRUPTS();
 					elog(DEBUG1, "TSQL TXN PG semantics : Rollback current transaction");
 					HoldPinnedPortals();
+					SPI_getCurrentInternalTxnMode(&spi_current_internal_xact_mode);
```

Comment:
Other wise Assert(current_spi_stack_depth == SPI_depth); fails for EXEC psql_interop_proc; (nested procedure tsql --> psql --> tsql)

This could indicate that we could enter this IF block with the top of SPI stack holding a tsql SPI connection, which is a internal xact = true, we do not want to mark this as false.

Reply:
removed all usage of spi_current_internal_xact_mode from catch block here. Could not find any case where we would need to protect spi from deletion here.
Abort Transaction will now cleanup SPI stack as well (until it finds an internal xact), which means we should never get a SPI leak in terminate batch.